examples/molecular_dynamics/lennard_jones_system.py

```diff
@@ -0,0 +1,280 @@
+                true_out = torch.tensor(
+                    (forces - force_mean) / force_sd, dtype=torch.float32
+                ).to(dist.device)
+
+                optimizer.zero_grad()
+                loss = F.l1_loss(out, true_out) + 0.001 * torch.mean(out).abs()
+                loss.backward()
+                optimizer.step()
+                scheduler.step()
+                log.log_minibatch({"Mini-batch loss": loss.detach()})
+            log.log_epoch({"Learning Rate": optimizer.param_groups[0]["lr"]})
+
+        if dist.rank == 0:
+            save_checkpoint(
+                "./checkpoints",
+                models=model,
+                optimizer=optimizer,
+                scheduler=scheduler,
+                epoch=epoch,
+            )
+
+            with LaunchLogger("valid", epoch=epoch) as log:
+                with torch.no_grad():
+                    model.eval()
+                    val_loss = 0
+
+                    forces_pair = []
+                    cosines = []
+                    for data in test_dataloader:
+
+                        pos = data[0][0]
+                        forces = data[1][0]
+                        pos, forces = prepare_input(
+                            pos,
+                            forces,
+                            box_size,
+                            rotation_matrix=None,
+                            add_random_noise=False,
+                        )
+
+                        # Create edges
+                        src, dst, edge_features = create_edges(
+                            pos, distance_threshold, box_size
+                        )
+                        g = dgl.graph((src, dst)).to(dist.device)
+                        node_fea = torch.ones(
+                            size=(pos.shape[0], cfg.model.hidden_dim_edge_encoder)
+                        ).to(dist.device)
+                        edge_fea = (
+                            torch.tensor(np.array(edge_features), dtype=torch.float32)
+                            .view(-1, 4)
+                            .to(dist.device)
+                        )
+
+                        out = model(node_fea, edge_fea, g)
+                        true_out = torch.tensor(
+                            (forces - force_mean) / force_sd, dtype=torch.float32
+                        ).to(dist.device)
+
+                        val_loss += F.mse_loss(out, true_out).detach()
+
+                        out_np = out.detach().cpu().numpy()
+                        true_out_np = true_out.detach().cpu().numpy()
+                        forces_pair.append((out_np, true_out_np))
+
+                        # angle computation
+                        dot_product = np.sum(out_np * true_out_np, axis=1)
+                        out_np_mag = np.linalg.norm(out_np, axis=1)
+                        true_out_np_mag = np.linalg.norm(true_out_np, axis=1)
+
+                        cosine = dot_product / (out_np_mag * true_out_np_mag)
```

Comment:
It would also be useful to have some commentary on the validation metrics.

Reply:
Added